### ISSUE TITLE
Warn when swDest is set to the webpack output directory

### ISSUE THREAD

**Issue by aganglada** [GoogleChrome/workbox#1363]
**Library Affected**:
workbox-webpack-plugin

**Browser & Platform**:
macOS High Sierra

**Issue or Feature Request Description**:
I'm using `workbox-webpack-plugin@^3.0.0` and trying to generate the service worker with this plugin setup:

```
new WorkboxPlugin.GenerateSW({
      chunks: Object.keys(base.entry),
      include: [/\.css$/, /\.js$/],
      swDest: path.resolve(__dirname, 'dist')
    })
```
and I'm getting this error when running `webpack-dev-server`, looks like the module is not able to write the file in the file system.

```
Error
    at MemoryFileSystem.writeFileSync (/node_modules/memory-fs/lib/MemoryFileSystem.js:211:9)
    at MemoryFileSystem.writeFile (/node_modules/memory-fs/lib/MemoryFileSystem.js:324:8)
    at writeOut (/node_modules/webpack/lib/Compiler.js:339:28)
    at Immediate.<anonymous> (/node_modules/memory-fs/lib/MemoryFileSystem.js:288:4)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
```

**Comment by djeeg** [comment]
InjectManifest is also hanging when defining swDest

Doesnt work

```
        new InjectManifest({
            importWorkboxFrom: 'local',
            swSrc: path.resolve(__dirname, "../client", "sw.js"),
            swDest: path.resolve(__dirname, "../../../output/webpack", process.env.APP_ENV, "sw.js"),
```

Does work

```
        new InjectManifest({
            importWorkboxFrom: 'local',
            swSrc: path.resolve(__dirname, "../client", "sw.js"),
            //swDest: 
```

**Comment by jeffposnick** [comment]
Quick question: are you using webpack v3 or v4?

**Comment by aganglada** [comment]
`"webpack": "^3.0.0",`

**Comment by jeffposnick** [comment]
Thanks for that info. I'll investigate.

**Comment by djeeg** [comment]
"webpack": "^3.8.1",

**Comment by jeffposnick** [comment]
Regarding @aganglada's configuration, the issue is that you're setting

```js
{
  ...
  swDest: path.resolve(__dirname, 'dist'),
}
```

Which means that you're attempting to **overwrite the output directory** with the contents of the generated service worker.

In general, you should be able to use `workbox-webpack-plugin` v3 with zero config, and you do not have to set `swDest`. It will default to `service-worker.js` in the output directory used by your config. (I need to double-check that that's accurately reflected in the docs.)  If you do choose to set `swDest`, set it to a path + filename to the new service worker file, not to a directory which will be overwritten.

(We currently [detect and warn against this](https://github.com/GoogleChrome/workbox/blob/7146dfe289789e1e7e6a9927e07ddfc6efa4755c/packages/workbox-build/src/lib/errors.js#L46-L47) when using the basic `workbox-build` node module, but when you add `webpack-dev-server` to the mix, with its in-memory filesystem, that detection logic would need to be customized.)

I'm still looking into @djeeg's issue, but I gather it might be related.

**Comment by aganglada** [comment]
Awesome, I will try that and let you know.
Thanks so much for your time.

**Comment by aganglada** [comment]
It worked! You rock.
Thank you so much!

**Comment by jeffposnick** [comment]
Great!

We cleaned up some of the documentation that might have led to confusion in https://github.com/google/WebFundamentals/pull/5928

I took a look at whether we can specifically detect when a developer passes in a `swDest` value that is the same as their output directory in the webpack plugin, and... it's a lot less straightforward that it is with `workbox-build`, due to having to account for webpack's support for both in-memory and real filesystems.

I'll leave this open to continue tracking that, but hopefully the documentation will make it clear that `swDest` is not required, and if provided, should include a filename.
